File: guides/micronaut-security-api-key/java/src/main/java/example/micronaut/AppController.java
package example.micronaut;

import io.micronaut.http.MediaType;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.Produces;
import io.micronaut.security.annotation.Secured;
import io.micronaut.security.rules.SecurityRule;

@Controller("/app") // <1>
class AppController {

    @Produces(MediaType.TEXT_PLAIN) // <2>
    @Secured(SecurityRule.IS_AUTHENTICATED) // <3>
    @Get // <4>
    String index() {
        return "Top Secret";
    }
}

File: guides/micronaut-security-api-key/python/src/example/micronaut/app_controller.py
from micronaut.http import MediaType
from micronaut.http.annotation import Get, Produces
from micronaut.security.annotation import Secured
from micronaut.security.rules import SecurityRule


# tag::clazz[]
@Produces(MediaType.TEXT_PLAIN)  # <2>
@Secured(SecurityRule.IS_AUTHENTICATED)  # <3>
@Get("/app")  # <1> <4>
def index() -> str:
    return "Top Secret"
# end::clazz[]
